src/homework/homeworks/myhashmap/MyHashMap.java

```diff
@@ -0,0 +1,142 @@
+        if (bucketsOccupied / array.length >= LOAD_FACTOR) {
+            Node<K, V>[] arrayNew = new Node[array.length * 2];
+            for (int i = 0; i < array.length; i++) {
+                arrayNew[array[i].key.hashCode() % arrayNew.length] = array[i];
+            }
+            array = arrayNew;
+        }
+        int hashCodeOfNode = key.hashCode();
+        int numerOfBucket = hashCodeOfNode % array.length;
+        Node<K, V> currentNode = array[numerOfBucket];
+        if (currentNode == null) {
+            currentNode = array[numerOfBucket] = new Node<K, V>(key, value);
+            bucketsOccupied++;
+        } else {
+            while (true) {
+                if (currentNode.key.equals(key)) {
+                    currentNode.value = value;
+                    break;
+                } else if (currentNode.nextNode == null) {
+                    currentNode.nextNode = new Node<K, V>(key, value);
+                    break;
+                } else {
+                    currentNode = currentNode.nextNode;
+                }
+            }
+        }
+
+        numberOfNodes++;
+    }
+
+    public void remove(K key) {
+        int hashCodeOfNode = key.hashCode();
+        int numberOfBucket = hashCodeOfNode % array.length;
+        Node<K, V> currentNode = array[numberOfBucket];
+        if (currentNode != null && currentNode.key.equals(key)) {
+            if (currentNode.nextNode == null) {
+                bucketsOccupied--;
+            }
+            array[numberOfBucket] = currentNode.nextNode;
+            numberOfNodes--;
+        } else if (currentNode != null) {
+            while (currentNode.nextNode != null) {
+                if (currentNode.nextNode.key.equals(key)) {
+                    currentNode.nextNode = currentNode.nextNode.nextNode;
+                    numberOfNodes--;
+                    break;
+                } else {
+                    currentNode = currentNode.nextNode;
+                }
+            }
+        }
+    }
+
+    public void clear() {
+        array = new Node[MAP_INITIAL_SIZE];
+        numberOfNodes = 0;
+        bucketsOccupied = 0;
+    }
+
+    public int size() {
+        return numberOfNodes;
+    }
+
+    public V get(K key) {
+        int hashCodeOfNode = key.hashCode();
+        int numberOfBucket = hashCodeOfNode % array.length;
+        Node<K, V> currentNode = array[numberOfBucket];
+        while (currentNode != null) {
+            if (currentNode.key.equals(key)) {
+                return currentNode.value;
+            } else {
+                currentNode = currentNode.nextNode;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String toString() {
+        return "MyHashMap{" +
+                "array=" + Arrays.toString(array) +
+                '}';
+    }
+
+    public static void main(String[] args) {
+        MyHashMap<Integer, String> myHashMap = new MyHashMap<>();
+        myHashMap.put(1, "1");
```

Comment:
Check case when entries with same keys are added and when entries with keys with equal hashcode added